WordPress/Classes/Networking/TaxonomyServiceRemoteREST.m

```diff
@@ -2,76 +2,148 @@
-              }
-          } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
-              if (failure) {
-                  failure(error);
-              }
-          }];
-}
-
 - (void)createCategory:(RemotePostCategory *)category
                success:(void (^)(RemotePostCategory *))success
                failure:(void (^)(NSError *))failure
 {
     NSParameterAssert(category.name.length > 0);
-    NSString *path = [NSString stringWithFormat:@"sites/%@/categories/new?context=edit", self.siteID];
-    NSString *requestUrl = [self pathForEndpoint:path
-                                     withVersion:ServiceRemoteRESTApiVersion_1_1];
     
     NSMutableDictionary *parameters = [NSMutableDictionary dictionary];
     parameters[@"name"] = category.name;
     if (category.parentID) {
         parameters[@"parent"] = category.parentID;
     }
+    
+    [self createTaxonomyWithType:TaxonomyServiceRemoteRESTCategoryTypeIdentifier
+                      parameters:parameters
+                         success:^(id responseObject) {
+                             RemotePostCategory *receivedCategory = [self remoteCategoryWithJSONDictionary:responseObject];
+                             if (success) {
+                                 success(receivedCategory);
+                             }
+                         } failure:failure];
+}
 
+- (void)getCategoriesWithSuccess:(void (^)(NSArray <RemotePostCategory *> *))success
+                         failure:(void (^)(NSError *))failure
+{
+    [self getCategoriesWithSuccess:success
+                      paging:nil
+                     failure:failure];
+}
+
+- (void)getCategoriesWithSuccess:(void (^)(NSArray <RemotePostCategory *> *categories))success
+                          paging:(RemoteTaxonomyPaging *)paging
+                         failure:(void (^)(NSError *error))failure
+{
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTCategoryTypeIdentifier
+                   parameters:[self parametersForPaging:paging]
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteCategoriesWithJSONArray:[responseObject arrayForKey:@"categories"]]);
+                          }
+                      } failure:failure];
+}
+
+- (void)searchCategoriesWithName:(NSString *)nameQuery
+                   success:(void (^)(NSArray <RemotePostCategory *> *tags))success
+                   failure:(void (^)(NSError *error))failure
+{
+    NSParameterAssert(nameQuery.length > 0);
+    [self getTaxonomyWithType:TaxonomyServiceRemoteRESTCategoryTypeIdentifier
+                   parameters:@{@"search": nameQuery}
+                      success:^(id responseObject) {
+                          if (success) {
+                              success([self remoteCategoriesWithJSONArray:[responseObject arrayForKey:@"categories"]]);
```

Comment:
Would it make sense to use `TaxonomyServiceRemoteRESTCategoryTypeIdentifier` instead of `@"categories"`?